org/postgresql/test/jdbc2/PGTimeTest.java

```diff
@@ -0,0 +1,268 @@
+/*-------------------------------------------------------------------------
+*
+* Copyright (c) 2004-2014, PostgreSQL Global Development Group
+*
+*
+*-------------------------------------------------------------------------
+*/
+package org.postgresql.test.jdbc2;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Time;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+import junit.framework.TestCase;
+
+import org.postgresql.test.TestUtil;
+import org.postgresql.util.PGInterval;
+import org.postgresql.util.PGTime;
+
+/**
+ * Tests {@link PGTime} in various scenarios including setTime, setObject for both
+ * <code>time with time zone</code> and <code>time without time zone</code> data types.
+ */
+public class PGTimeTest extends TestCase
+{
+    /** The name of the test table. */
+    private static final String TEST_TABLE = "testtime";
+
+    /** The database connection. */
+    private Connection con;
+
+    public PGTimeTest(String name)
+    {
+        super(name);
+    }
+
+    protected void setUp() throws Exception
+    {
+        con = TestUtil.openDB();
+        TestUtil.createTempTable(con, TEST_TABLE, "tm time, tz time with time zone");
+    }
+
+    protected void tearDown() throws Exception
+    {
+        TestUtil.dropTable(con, TEST_TABLE);
+        TestUtil.closeDB(con);
+    }
+
+    /**
+     * Tests that adding a <code>PGInterval</code> object to a
+     * <code>PGTime</code> object when performed as a casted string and object.
+     *
+     * @throws SQLException
+     *            if a JDBC or database problem occurs.
+     */
+    public void testTimeWithInterval() throws SQLException
+    {
+        Calendar cal = Calendar.getInstance();
+        cal.set(1970, 0, 1);
+
+        final long now = cal.getTimeInMillis();
+        verifyTimeWithInterval(new PGTime(now), new PGInterval(0, 0, 0, 1, 2, 3.14), true);
+        verifyTimeWithInterval(new PGTime(now), new PGInterval(0, 0, 0, 1, 2, 3.14), false);
+
+        verifyTimeWithInterval(new PGTime(now, Calendar.getInstance(TimeZone.getTimeZone("GMT"))), new PGInterval(0, 0, 0, 1, 2, 3.14), true);
+        verifyTimeWithInterval(new PGTime(now, Calendar.getInstance(TimeZone.getTimeZone("GMT"))), new PGInterval(0, 0, 0, 1, 2, 3.14), false);
+
+        verifyTimeWithInterval(new PGTime(now, Calendar.getInstance(TimeZone.getTimeZone("GMT+01:00"))), new PGInterval(0, 0, 0, 1, 2, 3.456), true);
+        verifyTimeWithInterval(new PGTime(now, Calendar.getInstance(TimeZone.getTimeZone("GMT+01:00"))), new PGInterval(0, 0, 0, 1, 2, 3.456), false);
```

Comment:
Please add "GMT-01:00" test as well. I have seen issues when "GMT+" works and "GMT-" does not (or vise versa)